google/cloud/aiplatform/featurestore/entity_type.py

```diff
@@ -1138,3 +1145,144 @@ def ingest_from_gcs(
+            response = self._featurestore_online_client.read_feature_values(
+                request=read_feature_values_request, metadata=request_metadata
+            )
+
+        elif isinstance(entity_ids, list):
+            streaming_read_feature_values_request = gca_featurestore_online_service.StreamingReadFeatureValuesRequest(
+                entity_type=self.resource_name,
+                entity_ids=entity_ids,
+                feature_selector=feature_selector,
+            )
+            response = [
+                response
+                for response in self._featurestore_online_client.streaming_read_feature_values(
+                    request=streaming_read_feature_values_request,
+                    metadata=request_metadata,
+                )
+            ]
+
+        return EntityType._load_read_feature_values_response_to_dataframe(
+            response=response
+        )
+
+    @staticmethod
+    def _load_read_feature_values_response_to_dataframe(
+        response: Union[
+            gca_featurestore_online_service.ReadFeatureValuesResponse,
+            List[gca_featurestore_online_service.ReadFeatureValuesResponse],
+        ],
+    ) -> "pd.DataFrame":  # noqa: F821 - skip check for undefined name 'pd'
+        """Loads read_feature_values_response to dataframe
+
+        Args:
+            response (Union[featurestore_online_service.ReadFeatureValuesResponse,
+                            List[featurestore_online_service.ReadFeatureValuesResponse]):
+                Required. featurestore_online_service.ReadFeatureValuesResponse from `read_feature_values`
+                or Iterable in list from `streaming_read_feature_values`.
+
+        Raises:
+            ImportError: If pandas is not installed when using this method.
+
+        Returns:
+            pd.DataFrame: response in DataFrame
+        )
+        """
+
+        try:
+            import pandas as pd
+        except ImportError:
+            raise ImportError(
+                f"Pandas is not installed. Please install pandas to use "
+                f"{EntityType._load_read_feature_values_response_to_dataframe.__name__}"
+            )
+
+        if isinstance(
+            response, gca_featurestore_online_service.ReadFeatureValuesResponse
+        ):
+            response_header = response.header
+            response_entity_view = [response.entity_view]
+        elif isinstance(response, list):
+            response_header = response[0].header
+            response_entity_view = [response.entity_view for response in response[1:]]
+
+        feature_ids = [
+            feature_descriptor.id
+            for feature_descriptor in response_header.feature_descriptors
+        ]
+        data = []
+
+        for entity_view in response_entity_view:
+            entity_id = entity_view.entity_id
+            entity_data = [entity_id]
+
+            for entity_view_data in entity_view.data:
+                if entity_view_data._pb.HasField("value"):
```

Comment:
`data` can be empty (i.e feature missing value). Should we append(None) in that case to make sure entity_data alignment?

Reply:
fixed this!